databroker/src/broker.rs

```diff
@@ -2010,6 +2014,7 @@ mod tests {
             .update_entries([(
                 id1,
                 EntryUpdate {
+                    id: Some(id1),
```

Comment:
Is this needed as we actually give `id1` a few line above as well. Like before we managed without the path.

Reply:
> Is this needed as we actually give id1 a few line above as well. Like before we managed without the path.

Yes, these changes should not be necessary.

Reply:
If `id` is not stored in `EntryUpdate` how is it going to return in the response `map` the corresponding `id` of the Datapoint?
It is not possible to use the request map ids order since the number of signals that might changed and will be returned to user might be dynamic right?. how could I get its ìd` then to build the response?

Reply:
> If `id` is not stored in `EntryUpdate` how is it going to return in the response 

By putting this information in the `ChangeNotification` instead.

You are right that the current implementation of `ChangeSubscription::notify` is (ab)using `EntryUpdate` to pass the path to the subscribers, even though it shouldn't.

```rust
// TODO: Perhaps make path optional
update.path = Some(entry.metadata.path.clone());
```

Compare with how `fn update(&mut self, id: i32, update: EntryUpdate)` and `QuerySubscription::notify` is not using `EntryUpdate` to communicate this information.

Reply:
Updated